src/Avalonia.Controls/TabControl.cs

```diff
@@ -145,6 +146,69 @@ bool IContentPresenterHost.RegisterContentPresenter(IContentPresenter presenter)
             return RegisterContentPresenter(presenter);
         }
 
+        protected override void OnContainersMaterialized(ItemContainerEventArgs e)
+        {
+            base.OnContainersMaterialized(e);
+
+            if (SelectedContent != null || SelectedIndex == -1)
+            {
+                return;
+            }
+
+            var container = (TabItem)ItemContainerGenerator.ContainerFromIndex(SelectedIndex);
+
+            if (container == null)
+            {
+                return;
+            }
+
+            UpdateSelectedContent(container);
+        }
+
+        private void UpdateSelectedContent(AvaloniaPropertyChangedEventArgs e)
+        {
+            var index = (int)e.NewValue;
+
+            if (index == -1)
+            {
+                SelectedContentTemplate = null;
+
+                SelectedContent = null;
+
+                return;
+            }
+
+            var container = (TabItem)ItemContainerGenerator.ContainerFromIndex(index);
+
+            if (container == null)
+            {
+                if (Items is AvaloniaList<object> items)
```

Comment:
I am curious - why do we need this special case?

Reply:
Y looks like this is no longer needed because we rely on the materialized event for the initial selection. If nothing breaks I will just remove this shortcut.